web/reports.go

```diff
@@ -586,8 +589,30 @@ func (web *Web) wpaKeysCsvReportHandler(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	settings, err := web.arena.Database.GetEventSettings()
+	if err != nil {
+		handleWebErr(w, err)
+		return
+	}
+
+	var filename string
+	tbaCode := settings.TbaEventCode
+	name := settings.Name
+	if tbaCode != "" {
+		// If we use the event code, scrub the leading digits. This creates
+		// the filenames that the WPA kiosk will expect, e.g. 2022CACC becomes
+		// CACC.
+		filename = eventCodeRegexp.FindStringSubmatch(tbaCode)[1]
+	} else if name != "" {
+		filename = name
+	} else {
+		filename = "default"
+	}
+
+	filename = filename + ".wpa_keys.csv"
+
 	w.Header().Set("Content-Type", "text/csv")
-	w.Header().Set("Content-Disposition", "attachment; filename=wpa_keys.csv")
```

Comment:
Given the exactly-four-characters requirement, would it be more user-friendly to just have some static or randomly generated name (e.g. `abcd.csv`) rather than the TBA code?

Reply:
Currently only event that uses Cheesy Arena (with advanced security) and doesn't have a 4 digit event code is Chezy Champs.

Reply:
I think a proper static name would be best, maybe "cakf.csv" and then the event name in the kiosk to "CAKF - My Event".